src/components/common/MembershipCardTermsModal.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Table as ChakraTable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@chakra-ui/react'
+import React from 'react'
+import { defineMessages, useIntl } from 'react-intl'
+import styled from 'styled-components'
+import { useMembershipCardTerms } from '../../hooks/membershipCardTerms'
+
+interface MembershipCardTermsModalProps {
+  isOpen: boolean
+  onClose: () => void
+  title: string
+  id: string
+}
+
+type ProductType = 'ActivityTicket' | 'ProgramPlan' | 'ProgramPackagePlan' | 'PodcastProgram'
+
+const messages = defineMessages({
+  ActivityTicket: { id: 'common.membershipCardTerm.activityTicket', defaultMessage: '活動' },
+  ProgramPlan: { id: 'common.membershipCardTerm.program', defaultMessage: '課程' },
+  ProgramPackagePlan: { id: 'common.membershipCardTerm.programPackage', defaultMessage: '課程組合' },
+  PodcastProgram: { id: 'common.membershipCardTerm.podcastProgram', defaultMessage: '廣播' },
+  cashDiscount: { id: 'common.membershipCardTerm.cashDiscount', defaultMessage: '折抵金額' },
+  percentageDiscount: { id: 'common.membershipCardTerm.percentageDiscount', defaultMessage: '折抵比例' },
+  generalDiscount: { id: 'common.membershipCardTerm.generalDiscount', defaultMessage: '折抵' },
+  type: { id: 'common.membershipCardTerm.type', defaultMessage: '類型' },
+  discountName: { id: 'common.membershipCardTerm.discountName', defaultMessage: '名稱' },
+  discountType: { id: 'common.membershipCardTerm.discountType', defaultMessage: '優惠類型' },
+  equityType: { id: 'common.membershipCardTerm.equityType', defaultMessage: '權益開通' },
+})
```

Comment:
could move to common transltaion, build a MembershipCardTermsModal defineMessages

Reply:
FYI: https://www.notion.so/kkshyu/Code-Review-c29ef181f60d489e939869373e77a065?pvs=4#f92f240916614d76851f009c95be27ce